### ISSUE TITLE
BadJpqlGrammarException with latest relase when using ORDER BY NULLS LAST

### ISSUE THREAD

**Issue by mavarazo** [spring-projects/spring-data-jpa#2962]
We faced today with the latest release from Spring Boot 3.1.0 an issue with a query which uses to order by nulls last.

```
@Query("select a, case when a.geaendertAm is null then a.erstelltAm else a.geaendertAm end as mutationAm from Element a" +
        " where a.erstelltDurch = :#{#erstelltDurch}" +
        " order by mutationAm desc nulls last")
    List<Element> findAllByErstelltDurchOrderByGeaendertAmDesc(String erstelltDurch);
```

The issue we have:
```
org.springframework.dao.InvalidDataAccessApiUsageException: org.springframework.data.jpa.repository.query.BadJpqlGrammarException: Line 1:378 mismatched input 'NULLS' expecting {<EOF>, ',', EXCEPT, FETCH, INTERSECT, LIMIT, OFFSET, UNION}; Bad JPQL grammar [select a, case when a.geaendertAm is null then a.erstelltAm else a.geaendertAm end as mutationAm from Element a where a.erstelltDurch = :__$synthetic$__1 order by mutationAm desc NULLS LAST]
```

Thanks for any advice to fix the issue

**Comment by gregturn** [comment]
Looks like we didn't implement all the right parts regarding `ORDER BY` the `NULLS (FIRST | LAST)` options.

**Comment by gregturn** [comment]
Merged to `main` and backported to `3.1.x`.

**Comment by gregturn** [comment]
Merged to `main` and backported to `3.1.x`.
